src/haddock/modules/analysis/ilrmsdmatrix/ilrmsd.py

```diff
@@ -0,0 +1,324 @@
+"""ilRMSD calculations."""
+import os
+from pathlib import Path
+
+import numpy as np
+
+from haddock import log
+from haddock.libs.libalign import (
+    calc_rmsd,
+    centroid,
+    get_atoms,
+    kabsch,
+    load_coords,
+    make_range,
+    )
+from haddock.modules.analysis.caprieval.capri import load_contacts
+
+
+class ContactJob:
+    """A Job dedicated to the fast retrieval of receptor contacts."""
+
+    def __init__(
+            self,
+            output,
+            params,
+            contact_obj):
+
+        log.info(f"core {contact_obj.core}, initialising Contact...")
+        self.params = params
+        self.output = output
+        self.contact_obj = contact_obj
+        log.info(f"core {contact_obj.core}, Contact initialised")
+
+    def run(self):
+        """Run this RMSDJob."""
+        log.info(f"core {self.contact_obj.core}, running ilRMSD...")
+        self.contact_obj.run()
+        self.contact_obj.output()
+        return
+
+
+class Contact:
+    """Contact class."""
+
+    def __init__(
+            self,
+            model_list,
+            output_name,
+            core,
+            path,
+            **params,
+            ):
+        """Initialise Contact class."""
+        self.model_list = model_list
+        self.output_name = output_name
+        self.core = core
+        self.path = path
+        self.atoms = {}
+        self.receptor_chain = params["params"]["receptor_chain"]
+        self.ligand_chains = params["params"]["ligand_chains"]
+        self.unique_rec_res = []
+        self.unique_lig_res = []
+        for m in model_list:
+            self.atoms.update(get_atoms(m))
+        
+    def run(self):
```

Comment:
question; you need to have this method for the schedulers to pick it up, right?

Reply:
exactly!